Seed swap

ebensonm opened the pull request:
Seed swap merge

ebensonm committed:
Working seed swap, need to add a query aspect

ebensonm committed:
hyperparameter functionality

ebensonm committed:
seed swap and merge fixes

ebensonm committed:
Merge branch 'pipeline_parser' of https://github.com/byu-dml/d3m-experimenter into seed_swap

ebensonm committed:
Primitive and seed swap updates

e13h commented on seed_swap.py:
```diff
@@ -1,26 +1,35 @@
+            run_pipeline_seed(pipeline, seed_num, **cli_args)
 
 def run_pipeline_seed(pipeline, seed, **cli_args):
     pipeline = Pipeline.from_json(string_or_file=pipeline)
```
What if you called `from_json_structure()` instead? Then you wouldn't need to even use `json.dumps()`. Is there a reason to convert the pipeline to a string?
```suggestion
    pipeline = Pipeline.from_json_structure(pipeline)
```

bjschoenfeld replied:
Yes, when working with pipelines, it should either be a file path or a json-like object.

e13h commented on seed_swap.py:
```diff
@@ -1,26 +1,35 @@
+    #get the relevant info from the query                               
+    pipeline, datasets_used, used_random_seeds = get_pipeline("f5478b6c-daad-4693-93c3-f073bae78ddf")
+    #now check if there are no used random seeds
     pipeline = json.dumps(pipeline, indent=4)
```
See my other comment
```suggestion
```

ebensonm committed:
Merge branch 'pipeline_parser' of https://github.com/byu-dml/d3m-experimenter into seed_swap

ebensonm committed:
Updated seed swap with query

Commit:
minor seed swap bug fixes

ebensonm committed:
queueing functionality beta_0

Commit:
Fix seed and pipeline run errors

Commit:
added some logging for debugging purposes

bjschoenfeld commented:
All files should be under the `experimenter/` directory.

bjschoenfeld commented on experimenter/run_pipeline.py:
```diff
@@ -137,19 +137,38 @@ def run(self, pipeline: Pipeline, metric_names: list = None) -> list:
-            volumes_dir=self.volumes_dir,
-            runtime_environment=runtime_environment,
-        )
+        if (self.random_seed is None):
+            all_scores, all_results = evaluate(
+                pipeline,
+                inputs,
+                data_pipeline=data_pipeline,
+                scoring_pipeline=scoring_pipeline,
+                problem_description=problem_description,
+                data_params=data_params,
+                metrics=metrics,
+                context=context,
+                volumes_dir=self.volumes_dir,
+                runtime_environment=runtime_environment,
+            )
+        #for seed change
+        else:
+            logger.warning('Random Seed: {}'.format(self.random_seed))
+            all_scores, all_results = evaluate(
+                pipeline,
+                inputs,
+                data_pipeline=data_pipeline,
+                scoring_pipeline=scoring_pipeline,
+                problem_description=problem_description,
+                data_params=data_params,
+                metrics=metrics,
+                context=context,
+                volumes_dir=self.volumes_dir,
+                runtime_environment=runtime_environment,
+                random_seed=self.random_seed,
+                scoring_random_seed=self.random_seed, 
+                data_random_seed=self.random_seed
+            )
+        
```
this is a lot of duplicated code. lets set variables based on conditions and then have only one call to `evaluate`.

bjschoenfeld commented on experimenter/execute_pipeline.py:
```diff
@@ -129,11 +129,12 @@ def handle_successful_pipeline_run(
-    if score["value"][0] == 0:
-        # F-SCORE was calculated wrong - quit and don't keep this run
-        return
+    #if score["value"][0] == 0:
+    #    # F-SCORE was calculated wrong - quit and don't keep this run
+    #    return
```
we should just remove this, not comment it out.

bjschoenfeld commented on experimenter/execute_pipeline.py:
```diff
@@ -129,11 +129,12 @@ def handle_successful_pipeline_run(
+    #    return
 
     print_pipeline(pipeline.to_json_structure(), score)
+    logger.warning("Succesfully printed pipeline")
```
this is an abuse of a warning. info would probably be better.

bjschoenfeld commented:
Its up to the author of the `pipeline_parser` branch to decide whether to merge this in and actually merge it. @evanphilipsmith @ebensonm.

The new code also needs some tests.

ebensonm committed:
Moved files to experimenter and fixed logging mishaps

Commit:
remove old query functions
Add util functions to check if a dataset or problem id exists

Commit:
fixed no data prep pipeline error

Commit:
Merge branch 'seed_swap' of https://github.com/byu-dml/d3m-experimenter into seed_swap

Commit:
Updates

ebensonm committed:
add unit test on simple pipeline

ebensonm committed:
added function to query for single pipeline test

Commit:
test bug fixes, still need to find a working test pipeline

Commit:
Working test

Commit:
add dynamic limiting to pipeline_generator()

ebensonm committed:
Now allows terminal call and includes method for using outside of terminal

Commit:
Minor bug fixes

Commit:
added ability to filter pipeline runs on submitter

ebensonm committed:
Removed duplicate code for testing and single pipeline_id submission

Commit:
Fixed problems with test and single pipeline_id changes

Commit:
begin adding scoring and data prep pipelines

Commit:
Added pipelines

Commit:
modify pipeline generator to limit results based on number of random seeds
scan_pipeline_runs() now returns a the random seeds used for every pipeline-problem-dataset tuple it finds

Commit:
change default num_pipelines parameter and minor fixes

bjschoenfeld commented:
@ebensonm I think this PR is now obsolete. If so, can you close it?
